domain/lib/src/model/document/document.dart

```diff
@@ -61,22 +63,31 @@ class Document extends Equatable {
       this.sha256sum,
       this.hasThumbnail,
       this.shared,
-      this.mediaType);
+      this.mediaType,
+      {
+        this.localPath,
+        this.syncOfflineState = SyncOfflineState.NONE
+      }
```

Comment:
IMO, it is not the properties of core entity `Document`. How about create another model for this which is extended from `Document`?

Reply:
That is too easy. It's not the object we need to save so it doesn't need to be bound here.